app/core/models/ocelot/stanzas/CalculationStanza.scala

```diff
@@ -151,21 +149,74 @@ case class FloorOperation(left: String, right: String, label: String) extends Op
 
 }
 
-case class Calculation(override val next: Seq[String], calcs: Seq[Operation]) extends Stanza with Evaluate {
+object Operation {
+  implicit val reads: Reads[Operation] = (js: JsValue) => {
+    (js \ "type").validate[String] match {
+      case err @ JsError(_) => err
+      case JsSuccess(typ, _) => typ match {
+        case "add" => js.validate[AddOperation]
+        case "sub" => js.validate[SubtractOperation]
+        case "ceil" => js.validate[CeilingOperation]
+        case "flr" => js.validate[FloorOperation]
+        case typeName => JsError(JsonValidationError(Seq("ChoiceTest"), typeName))
```

Comment:
Operation rather than ChoiceTest?

Reply:
yes as Calculation has next and Operation sequences, ChoiceTest is on CalculationStanza

Reply:
Ah realised you meant in the error message :), will update